ownCloud/Settings/LogFilesViewController.swift

```diff
@@ -0,0 +1,243 @@
+					self?.shareLog(with: name)
+				}
+			}
+		}
+
+		return cell!
+	}
+
+	override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		tableView.deselectRow(at: indexPath, animated: true)
+	}
+
+	override func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
+
+		return [
+			UITableViewRowAction(style: .destructive, title: "Delete".localized, handler: { [weak self] (_, indexPath) in
+				let logEntry = self?.logFileEntries[indexPath.row]
+				if let name = logEntry?.name {
+					self?.removeLog(with: name, indexPath: indexPath)
+				}
+			})
+		]
+	}
+
+	// MARK: - Private Helpers
+
+	private func populateLogFileList() {
+		guard let logFileWriter = OCLogger.shared.writer(withIdentifier: .writerFile) as? OCLogFileWriter else { return }
+		guard let appGroupLogsURL = OCAppIdentity.shared.appGroupLogsContainerURL else { return }
+		guard let logFiles = logFileWriter.logFiles() else { return }
+
+		for fileName in logFiles {
+			if let name = fileName as? String {
+				let logURL = appGroupLogsURL.appendingPathComponent(name)
+
+				var fileSize:Int64?
+				var creationDate:Date?
+
+				if let attributes = try? FileManager.default.attributesOfItem(atPath: logURL.path) {
+					creationDate = attributes[FileAttributeKey.creationDate] as? Date
+					fileSize = (attributes[FileAttributeKey.size] as? NSNumber)?.int64Value
+				}
+
+				let logEntry = LogEntry(name: name, size: fileSize, creationDate: creationDate)
+				self.logFileEntries.append(logEntry)
+			}
+		}
+
+		self.tableView.reloadData()
+	}
+
+	private func shareLog(with fileName:String) {
+		guard let appGroupLogsURL = OCAppIdentity.shared.appGroupLogsContainerURL else { return }
+		let shareableFileName = fileName + ".txt"
+		let shareableLogURL = FileManager.default.temporaryDirectory.appendingPathComponent(shareableFileName)
+		let logURL = appGroupLogsURL.appendingPathComponent(fileName)
+
+		do {
+			if FileManager.default.fileExists(atPath: shareableLogURL.path) {
+				try FileManager.default.removeItem(at: shareableLogURL)
+			}
+
+			try FileManager.default.copyItem(at: logURL, to: shareableLogURL)
+		} catch {
+		}
+
+		let shareViewController = UIActivityViewController(activityItems: [shareableLogURL], applicationActivities:nil)
+		shareViewController.completionWithItemsHandler = { (_, _, _, _) in
+			do {
+				try FileManager.default.removeItem(at: shareableLogURL)
+			} catch {
+			}
+		}
+
+		if UIDevice.current.isIpad() {
+			shareViewController.popoverPresentationController?.sourceView = self.view
+		}
+		self.present(shareViewController, animated: true, completion: nil)
+	}
+
+	private func removeLog(with fileName:String, indexPath:IndexPath) {
+		guard let appGroupLogsURL = OCAppIdentity.shared.appGroupLogsContainerURL else { return }
+		let logURL = appGroupLogsURL.appendingPathComponent(fileName)
+		do {
+			try FileManager.default.removeItem(at: logURL)
```

Comment:
Removal, too, should be managed through the same method both in the SDK and in the app, f.ex. through a `-[OCLogWriter deleteLog:(OCLogFileRecord *)log completionHandler:(void(^)(NSError *error))completionHandler]` method. As it is, the actual implementation to delete an individual log file exists twice - once in the app and once in the SDK.

Reply:
@felix-schwarz done